File: src/test/java/com/flexpoker/table/command/aggregate/generic/ResumeTableTest.java
package com.flexpoker.table.command.aggregate.generic;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import java.util.UUID;

import org.junit.jupiter.api.Test;

import com.flexpoker.exception.FlexPokerException;
import com.flexpoker.table.command.aggregate.testhelpers.TableTestUtils;
import com.flexpoker.table.command.events.TableResumedEvent;

public class ResumeTableTest {

    @Test
    void testResumeSuccess() {
        var table = TableTestUtils.createBasicTable(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID());
        table.pause();
        table.resume();
        assertEquals(TableResumedEvent.class, table.fetchNewEvents().get(table.fetchNewEvents().size() - 1).getClass());
    }

    @Test
    void testResumeOnActiveTable() {
        var table = TableTestUtils.createBasicTable(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID());
        assertThrows(FlexPokerException.class, () -> table.resume());
    }

    @Test
    void testResumeTwice() {
        var table = TableTestUtils.createBasicTable(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID());
        table.pause();
        table.resume();
        assertThrows(FlexPokerException.class, () -> table.resume());
    }

}

File: src/test/kotlin/com/flexpoker/table/command/aggregate/generic/ResumeTableTest.kt
package com.flexpoker.table.command.aggregate.generic

import com.flexpoker.exception.FlexPokerException
import com.flexpoker.table.command.aggregate.TableState
import com.flexpoker.table.command.aggregate.eventproducers.pause
import com.flexpoker.table.command.aggregate.eventproducers.resume
import com.flexpoker.table.command.aggregate.testhelpers.createBasicTable
import com.flexpoker.table.command.events.TableEvent
import com.flexpoker.table.command.events.TableResumedEvent
import com.flexpoker.test.util.EventProducerApplierBuilder
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import java.util.UUID

class ResumeTableTest {

    @Test
    fun testResumeSuccess() {
        val initState = createBasicTable(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID())
        val (_, events) = EventProducerApplierBuilder<TableState, TableEvent>()
            .initState(initState)
            .andRun { pause(it) }
            .andRun { resume(it) }
            .run()
        assertEquals(2, events.size)
        assertEquals(TableResumedEvent::class.java, events[1].javaClass)
    }

    @Test
    fun testResumeOnActiveTable() {
        val initState = createBasicTable(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID())
        assertThrows(FlexPokerException::class.java) { resume(initState) }
    }

    @Test
    fun testResumeTwice() {
        val initState = createBasicTable(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID())
        EventProducerApplierBuilder<TableState, TableEvent>()
            .initState(initState)
            .andRun { pause(it) }
            .andRun { resume(it) }
            .andRunThrows(FlexPokerException::class.java) { resume(it) }
            .run()
    }

}